src/main/java/io/logz/guice/jersey/JerseyServer.java

```diff
@@ -55,25 +58,28 @@ private void configureServer() {
             server.addConnector(connector);
         });
 
-        WebAppContext webAppContext = new WebAppContext();
-        webAppContext.setServer(server);
+        if (websocketConfiguration != null && "/".equals(jerseyConfiguration.getContextPath())) {
+            throw new InvalidWebSocketException(
+                    "The websocket context path cannot be the same as jersey context path. " +
+                            "Either set a context path for the rest API or disable the websocket feature. " +
+                            "Also make sure that a REST resource does not have the same path as a websocket path.");
+        }
 
-        webAppContext.addFilter(GuiceFilter.class, "/*", EnumSet.allOf(DispatcherType.class));
+        Optional<ContextHandler> webAppContext = new RestContextConfigurator(server, injectorSupplier)
+                .configure(jerseyConfiguration);
 
-        ServletHolder holder = new ServletHolder(ServletContainer.class);
-        holder.setInitParameter("javax.ws.rs.Application", GuiceJerseyResourceConfig.class.getName());
+        Optional<ContextHandler> websocketContext = new WebsocketContextConfigurator(server, injectorSupplier)
```

Comment:
Also here you only want to create handler when you supplied with `websocketConfiguration` so let's check here if it's not null and then create the context: 
```java
if (websocketConfiguration != null) {
    ContextHandler wsContextHandler = WebsocketContextHandlerBuilder.build(injectorSupplier, websocketConfiguration);
    handlers.add(wsContextHandler);
}
```